frontend/public/components/utils/file-input.tsx

```diff
@@ -21,19 +31,57 @@ export class FileInput extends React.Component<FileInputProps, FileInputState> {
     };
     reader.readAsText(file, 'UTF-8');
   }
+  onFileDrop(event) {
+    this.setState({ dropzoneActive: false });
+    this.readFile(event[0]);
+  }
+  onFileUpload(event) {
+    this.readFile(event.target.files[0]);
+  }
+  activeDragZone() {
+    this.setState({ dropzoneActive: true });
+  }
+  deactiveDragZone() {
+    this.setState({ dropzoneActive: false });
+  }
   render() {
-    return <div className="input-group">
-      <input type="text"
-        className="form-control"
-        value={this.state.inputFileName}
-        readOnly
-        disabled />
-      <span className="input-group-btn">
-        <span className="btn btn-default co-btn-file">
-          Browse&hellip;
-          <input type="file" onChange={this.onFileChange} className="form-control"/>
-        </span>
-      </span>
+    const id = this.props.id;
+    return <div className="form-group">
+      <Dropzone
+        onDrop={this.onFileDrop}
+        disableClick
+        style={{position: 'relative'}}
+        onDragEnter={this.activeDragZone}
+        onDragLeave={this.deactiveDragZone}>
+        { this.state.dropzoneActive && <div className="co-file-dropzone"><div className="drop-text">Drop file here</div></div> }
+        <div className="form-group">
+          <label className="control-label" htmlFor={id}>{this.props.label}</label>
+          <div className="modal-body__field">
+            <div className="input-group">
+              <input type="text"
+                className="form-control"
+                value={this.state.inputFileName}
+                readOnly
+                disabled />
+              <span className="input-group-btn">
+                <span className="btn btn-default co-btn-file">
+                  Browse&hellip;
+                  <input type="file" onChange={this.onFileUpload} className="form-control"/>
+                </span>
+              </span>
+            </div>
+            <p className="help-block">{this.props.inputFieldHelpText}</p>
+            <textarea className="form-control co-create-secret-form__textarea"
+              id={id}
+              onChange={this.onDataChange}
+              value={this.state.inputFileData}
+              aria-describedby={`${id}-help`}
```

Comment:
Looks like this should be

```html
aria-describedby="ssh-privatekey-help"
```

Reply:
No the `help-block` paragraph below should be set to `id={`${id}-help`}`